simulariumio/tests/test_agents_from_lists.py

```diff
@@ -0,0 +1,344 @@
+        radii=scale_factor * np.array(radii),
+    )
+
+    result = AgentData.from_lists(list_data, scale_factor)
+    assert expected_results == result
+
+
+def test_jagged_data_subpoints():
+    times = [0.0, 1.00001, 2.00001]
+    n_agents = [3, 3, 3]
+    viz_types = [
+        [1001.0, 1001.0, 1001.0],
+        [1001.0, 1001.0, 1001.0],
+        [1001.0, 1001.0, 1001.0],
+    ]
+    unique_ids = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+    types = [["H", "A", "C"], ["L", "D", "A"], ["K", "K", "A"]]
+    positions = [
+        [[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]],
+        [[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]],
+        [[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]],
+    ]
+    radii = [[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]]
+    n_subpoints = [[9, 12, 6], [9, 9, 6], [9, 6, 6]]
+    jagged_subpoints = [
+        [
+            [
+                -243.14059805,
+                207.75566987,
+                -95.33921063,
+                -20.54663446,
+                475.97201603,
+                14.43506311,
+                -76.45581828,
+                -97.31170699,
+                -144.30184731,
+                0.0,
+            ],
+            [
+                108.28447939,
+                175.55049775,
+                -274.34792273,
+                13.44237701,
+                258.21483663,
+                -65.05452787,
+                224.55922362,
+                -455.56482869,
+                -351.23389958,
+                -286.95502659,
+                330.12683064,
+                183.79420473,
+            ],
+            [
+                49.76236816,
+                -353.11708296,
+                226.84570983,
+                -234.5462914,
+                105.46507228,
+                17.16552317,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+            ],
+        ],
+        [
+            [
+                -442.27202981,
+                202.83568625,
+                -262.13407113,
+                -372.23130078,
+                217.21997368,
+                404.88561338,
+                171.37918011,
+                205.80515525,
+                -65.95336727,
+                0.0,
+                0.0,
+                0.0,
+            ],
+            [
+                245.9111405,
+                372.15936027,
+                -261.94702214,
+                3.50037066,
+                441.92904046,
+                321.75701298,
+                146.23928574,
+                -315.3241668,
+                82.00405173,
+            ],
+            [
+                104.82606074,
+                -413.76671598,
+                366.66127719,
+                136.7228888,
+                -210.69313998,
+                -465.59967482,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
+                0.0,
```

Comment:
does the length of these zeros matter?

Reply:
Yes and no. We only care about the non-zero values, those represent the actual subpoint data, but since numpy doesn't support jagged lists, each numpy subpoint array that will eventually be generated for the `AgentData` object will have exactly 12 elements. For this test, I just added various zeros (keeping the number of elements <= 12) to make sure there aren't any weird edge cases when we're filling in the zeros